src/test/java/com/dsrm/dsrmbackend/AuthControllerTests.java

```diff
@@ -81,6 +90,143 @@ public void authorizeWithCorrectData() throws Exception {
         tokenPayload = new String(decoder.decode(refreshToken[1]));
         username = JsonPath.read(tokenPayload, "$.username");
         assertEquals("test01@wp.pl", username);
+    }
+
+    @Test
+    public void refreshValidToken() throws Exception {
+        LoginDetailsRequestDTO userdata = new LoginDetailsRequestDTO();
+        userdata.setUsername("test01@wp.pl");
+        userdata.setPassword("zaq1@WSX");
+
+        JwtResponse tokens = authService.authenticateUser(userdata);
+
+        String refreshToken = tokens.getRefreshToken();
+        MvcResult refreshResult = this.mockMvc.perform(put("/refresh")
+                        .header("Authorization", refreshToken))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        String accessToken = JsonPath.read(refreshResult.getResponse().getContentAsString(), "$.accessToken").toString();
+        refreshToken = JsonPath.read(refreshResult.getResponse().getContentAsString(), "$.refreshToken").toString();
+
+        Claims claims;
+        try {
+            claims = (Claims) Jwts.parserBuilder()
+                    .setSigningKey(authService.getSigningKey())
+                    .build()
+                    .parse(accessToken)
+                    .getBody();
+        } catch (Exception e) {
```

Comment:
I wouldn't catch this exception. If exception will be thrown then test will fail.